commit-graph.c

```diff
@@ -747,7 +765,18 @@ static void fill_commit_graph_info(struct commit *item, struct commit_graph *g,
 	date_low = get_be32(commit_data + g->hash_len + 12);
 	item->date = (timestamp_t)((date_high << 32) | date_low);
 
-	graph_data->generation = get_be32(commit_data + g->hash_len + 8) >> 2;
+	if (g->chunk_generation_data) {
+		offset = (timestamp_t) get_be32(g->chunk_generation_data + sizeof(uint32_t) * lex_index);
+
+		if (offset & CORRECTED_COMMIT_DATE_OFFSET_OVERFLOW) {
+			offset_pos = offset - CORRECTED_COMMIT_DATE_OFFSET_OVERFLOW;
+			graph_data->generation = get_be64(g->chunk_generation_data_overflow + sizeof(timestamp_t) * offset_pos);
```

Comment:
Is `timestamp_t` always 64-bits? Perhaps we should use a constant or `sizeof(uint64_t)` here?

Reply:
`timestamp_t` is not always 64-bits, it's implementation specific. Since the neighbouring code prefers to use constant (for example: `date_high = get_be32(commit_data + g->hash_len + 8) 0x3)`, I will move ahead with using constant.